src/solvers/refinement/string_refinement.cpp

```diff
@@ -1665,6 +1715,9 @@ bool string_refinementt::is_axiom_sat(
   case decision_proceduret::resultt::D_UNSATISFIABLE:
     return false;
   default:
-    throw "failure in checking axiom";
+    error() << "failure in checking axiom" << eom;
```

Comment:
But in the future where an invariant can be caught, is there a guarantee that the message will be seen?